packages/rich-text/src/plugins/List/utils.ts

```diff
@@ -37,8 +37,21 @@ export const isNonEmptyListItem = (editor: PlateEditor, [, path]: NodeEntry) =>
   return listItemChildren.length !== 0;
 };
 
+export const firstNodeIsNotList = (_editor: PlateEditor, [node]: NodeEntry<CustomElement>) => {
+  const firstNode = node.children?.[0];
+
+  return !!firstNode && !Text.isText(firstNode) && !isList(firstNode);
+};
+
 export const insertParagraphAsChild = (editor: PlateEditor, [, path]: NodeEntry) => {
   Transforms.insertNodes(editor, [{ type: BLOCKS.PARAGRAPH, data: {}, children: [{ text: '' }] }], {
     at: path.concat([0]),
   });
 };
+
+export const replaceNodeWithListItems = (editor, entry) => {
+  const [node, path] = entry;
+
+  Transforms.removeNodes(editor, { at: path });
+  Transforms.insertNodes(editor, node.children[0].children, { at: path });
```

Comment:
In this case, `node` is the LI and it can only have two children, the `LIC` or another list such as `UL` and `OL`. If we know it doesn't have the `LIC`, that can be a paragraph, a heading, a embedded, etc, the only thing left if the list and we are checking if we do have a list as a first child here when doing `validNode: firstNodeIsNotList`.

I don't think the order will be changed because we always have `LI > LIC + UL`, otherwise the rendering would be different since this order will generate the final HTML.

> What happens to other children?

Which children? Like I said, we only have `UL > LI > LIC + UL?`, so a `UL` can only have `LI` as a child regardless of the nesting level, so I don't think we are losing anything here.

Reply:
I meant something like this:

```
<ul>
  <li>
    <ul> ... </ul>
    <p> other li child</p>
  </li>
</ul>
```

That's technically a valid structure, right? the code in `firstNodeIsNotList` will reject this case so we will kick the normalization. Yes, the node here is the `li` so this code:

```js
Transforms.insertNodes(editor, node.children[0].children, { at: path });
```

Takes the child `ul` and moves it in-place of the original `li` but it seems to be me the second child (the `p` in this case) will get lost.

Reply:
I can check if the list item has only 1 child, like `node.children.length === 1` and it's a list. What you think?

Reply:
Yes, that's better